distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java

```diff
@@ -180,12 +180,14 @@ static String agentCommandLineOption(Path agentJar, Path tmpPropertiesFile) {
         return "-javaagent:" + agentJar + "=c=" + tmpPropertiesFile;
     }
 
-    private static void extractSecureSettings(KeyStoreWrapper keystore, Map<String, String> propertiesMap) {
-        final Set<String> settingNames = keystore.getSettingNames();
+    private static void extractSecureSettings(SecureSettings secrets, Map<String, String> propertiesMap) {
+        final Set<String> settingNames = secrets.getSettingNames();
         for (String key : List.of("api_key", "secret_token")) {
             if (settingNames.contains("tracing.apm." + key)) {
-                try (SecureString token = keystore.getString("tracing.apm." + key)) {
+                try (SecureString token = secrets.getString("tracing.apm." + key)) {
                     propertiesMap.put(key, token.toString());
+                } catch (GeneralSecurityException gse) {
```

Comment:
This exception in the signature is leftover from an older implementation of the keystore. I think we can simply remove it as a checked exception in the SecureSettings interface?

Reply:
Done.